src/Manager.cc

```diff
@@ -295,9 +297,11 @@ ManagerPrivate::ManagerPrivate()
       std::bind(&ManagerPrivate::OnSigIntTerm, this, std::placeholders::_1));
 
   // Initialize semaphore
-  this->stoppedChildSem = sem_open(kSemaphoreName, O_CREAT, 0644, 1);
+  this->stoppedChildSemName = std::string("ign-launch-") + common::Uuid().String();
+  this->stoppedChildSem = sem_open(this->stoppedChildSemName.c_str(), O_CREAT, 0644, 1);
```

Comment:
This causes issues in MacOS CI, as the new name is too long.  I'm not sure exactly what `SEM_NAME_LEN` is on MacOS, but a good guess would be 31 from something like: https://chromium.googlesource.com/native_client/src/native_client/+/d6d53e83fffcc64b6e0d52acc31c63e0b5a60186/src/shared/platform/osx/nacl_semaphore.c#52

Reply:
Maybe something like `mktemp` would be more suitable here.

Reply:
Okay, I posted an alternative solution which should generate about 22-char filenames.

I haven't found a nice way to strongly seed the Rand() generator, so the seeding part is a bit ugly. `mktemp` doesn't sound good to me - it still seems to me that this pseudorandom magic would be faster than creating a file just for reading its name and closing it.

Reply:
I needed to add a private dependency of libignition-launch1.so on libignition-math6.so. I hope it's not considered as ABI break.